pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -145,6 +150,39 @@ func (r *AtlasClusterReconciler) Reconcile(context context.Context, req ctrl.Req
 	return workflow.OK().ReconcileResult(), nil
 }
 
+func (r *AtlasClusterReconciler) verifyNonTenantCase(cluster *mdbv1.AtlasCluster) {
+	var pSettings *mdbv1.ProviderSettingsSpec
+	var clusterType string
+	if cluster.Spec.ClusterSpec != nil {
+		if cluster.Spec.ClusterSpec.ProviderSettings == nil {
+			return
+		}
+		pSettings = cluster.Spec.ClusterSpec.ProviderSettings
+		clusterType = "TENANT"
+	}
+
+	if cluster.Spec.ServerlessSpec != nil {
+		if cluster.Spec.ServerlessSpec.ProviderSettings == nil {
+			return
+		}
+		pSettings = cluster.Spec.ServerlessSpec.ProviderSettings
+		clusterType = "SERVERLESS"
+	}
+
+	modifyProviderSettings(pSettings, clusterType)
+}
+
+func modifyProviderSettings(pSettings *mdbv1.ProviderSettingsSpec, clusterType string) {
+	if pSettings == nil || string(pSettings.ProviderName) == clusterType {
+		return
+	}
+	switch pSettings.InstanceSizeName {
+	case "M0", "M2", "M5":
```

Comment:
I think for "serverless" you have to do the swap for every instance size

Reply:
Yes, thanks!